### PR TITLE
Misc. fixes to dlg_contentstore.lua

### PR DESCRIPTION
**Pull request by ClobberXD**
See https://github.com/minetest/minetest/issues/7447#issuecomment-402906214 for more details
- Add warning when no packages are fetched
- Add missing `core.formspec_escape` to `get_screenshot`
- Internal name of the "Update" button was the same as "Install". Rename to `update_` which is suffixed by package number.

capture

Strangely, applying these changes resulted in the packages being correctly retrieved _almost_ every time. o_O

### CODE REVIEW

File: builtin/mainmenu/dlg_contentstore.lua [hunk 1]
```diff
@@ -351,20 +351,24 @@ function store.get_formspec()
...
-		filter_type,
-		"]",
-		-- "textlist[0,1;2.4,5.6;a;",
-		-- table.concat(taglist, ","),
-		-- "]"
+		"size[12,6.5;true]"
 	}
 
+	if #store.packages ~= 0 then
+		formspec[#formspec + 1] = "field[0.3,0.1;10.2,1;search_string;;"
```
Comment (rubenwardy) [inline builtin/mainmenu/dlg_contentstore.lua]: it would be better to initialise the list here. ie:

```lua
local formspec
if #store.packages ~= 0 then
    formspec = {
        ---
    }
else
    formspec = {
        ---
    }
end
```

File: builtin/mainmenu/dlg_contentstore.lua [hunk 2]
```diff
@@ -396,7 +400,7 @@ function store.get_formspec()
 			formspec[#formspec + 1] = fgettext("Install")
 			formspec[#formspec + 1] = "]"
 		elseif package.installed_release < package.release then
-			formspec[#formspec + 1] = "button[6,0;1.5,1;install_"
+			formspec[#formspec + 1] = "button[6,0;1.5,1;update_"
```
Comment (rubenwardy) [inline builtin/mainmenu/dlg_contentstore.lua]: wait, what? :O

Comment (ClobberXD) [reply builtin/mainmenu/dlg_contentstore.lua]: _Someone_ probably missed that... :)

File: builtin/mainmenu/dlg_contentstore.lua (hunk 1 above)
Comment (ClobberXD) [reply builtin/mainmenu/dlg_contentstore.lua]: Since `size[]` is required irrespective of whether packages can be retrieved or not, which is why I initialized `formspec` with a `size[]` element. Should I add the `size[]` element to both the initializers instead?

It'd actually look more cleaner than adding `formspec[#formspec + 1]` at the start of every line...

Comment (rubenwardy) [reply builtin/mainmenu/dlg_contentstore.lua]: Yes. It's also more efficient

**Comment by ClobberXD** [comment]
Is it OK if I attempt to also fix #7408 and #7305 in this PR? Both these issues complement each other, and here's the fix I have in mind:
- Remove short desc. from the main page (it's also displayed in the "View" dialog anyway)
- Place `Install` and `View` buttons below the title.

**Comment by ClobberXD** [comment]
@paramat, I only tweaked a couple of formspec stuff, and did not modify the fetching code in any way. I still did encounter the issue of no packages being retrieved, but much rarer. Co-incidence, more likely :)

Fixed button height continues to bother me. Since that won't be attended to, I'll do the following:
- Increase the height of the content store dialog to space out the elements
  - Doing this would allow me to place buttons below the title, fixing #7408
- Reduce packages per page from 5 to 4
  - And this should fix #7385 

Shall I go ahead?

**Comment by paramat** [comment]
Keep in mind the default game window size is 1024x600, so you need to test at height 600,this is considered the rough minimum window height we support.
Both your suggestions seem good but check with @rubenwardy first.

**Comment by paramat** [comment]
See http://irc.minetest.net/minetest-dev/2018-09-29#i_5416132

**Comment by ClobberXD** [comment]
I agree that my solutions weren't the best. Would someone please suggest something better?

Also, should this be merged without including fixes for #7408 and #7385? See first comment for the original intention of this PR.

@paramat @rubenwardy

**Comment by paramat** [comment]
I think leave format changes to another PR in order to not delay this one.

**Comment by paramat** [comment]
#7385 is being attended to by #7806

**Comment by ClobberXD** [comment]
What remains in this PR is just a couple of fixes and improvements unrelated to #7385 and #7408. Could this be merged?

**Comment by ClobberXD** [comment]
Accidental close, sorry :)

**Comment by ClobberXD** [comment]
Yes, good point

File: builtin/mainmenu/dlg_contentstore.lua (hunk 1 above)
Comment (paramat) [reply builtin/mainmenu/dlg_contentstore.lua]: Seems unresolved and not done.

Comment (ClobberXD) [reply builtin/mainmenu/dlg_contentstore.lua]: I'm sorry, my bad. Will fix ASAP.

Comment (ClobberXD) [reply builtin/mainmenu/dlg_contentstore.lua]: Done. @paramat

**Comment by ClobberXD** [comment]
Added a screenshot of the warning label to the first post.

File: builtin/mainmenu/dlg_contentstore.lua [hunk 3]
```diff
@@ -356,21 +356,29 @@ function store.get_formspec()
...
+	local formspec
+	if #store.packages ~= 0 then
+		formspec = {
+			"size[12,7;true]",
+			"position[0.5,0.55]",
+			"field[0.2,0.1;7.8,1;search_string;;",
+			core.formspec_escape(search_string), "]",
+			"field_close_on_enter[search_string;false]",
+			"button[7.7,-0.2;2,1;search;",
+			fgettext("Search") .. "]",
```
Comment (paramat) [inline builtin/mainmenu/dlg_contentstore.lua]: '..' has been added, is it necessary? Just wondering.

**Comment by paramat** [comment]
Minor question, otherwise +1.

**Comment by paramat** [comment]
+1 once that '..' is fixed, then i'll merge this.

**Comment by paramat** [comment]
:+1:
